protocol/x/revshare/keeper/revshare_test.go

```diff
@@ -75,3 +78,92 @@ func TestCreateNewMarketRevShare(t *testing.T) {
 	expectedExpirationTs := ctx.BlockTime().Unix() + 240*24*60*60
 	require.Equal(t, details.ExpirationTs, uint64(expectedExpirationTs))
 }
+
+func TestGetMarketMapperRevenueShareForMarket(t *testing.T) {
+	tests := map[string]struct {
+		revShareParams     types.MarketMapperRevenueShareParams
+		marketId           uint32
+		expirationDelta    int64
+		setRevShareDetails bool
+
+		// expected
+		expectedMarketMapperAddr sdk.AccAddress
+		expectedRevenueSharePpm  uint32
+		expectedErr              error
+	}{
+		"valid market": {
+			revShareParams: types.MarketMapperRevenueShareParams{
+				Address:         constants.AliceAccAddress.String(),
+				RevenueSharePpm: 100_000, // 10%
+				ValidDays:       0,
+			},
+			marketId:           42,
+			expirationDelta:    10,
+			setRevShareDetails: true,
+
+			expectedMarketMapperAddr: constants.AliceAccAddress,
+			expectedRevenueSharePpm:  100_000,
+		},
+		"invalid market": {
+			revShareParams: types.MarketMapperRevenueShareParams{
+				Address:         constants.AliceAccAddress.String(),
+				RevenueSharePpm: 100_000, // 10%
+				ValidDays:       0,
+			},
+			marketId:           42,
+			setRevShareDetails: false,
+
+			expectedErr: types.ErrMarketMapperRevShareDetailsNotFound,
+		},
+		// TODO: investigate why tApp blocktime doesn't translate to ctx.BlockTime()
```

Comment:
Sure. This isn't important at the moment because this test case and code is covered by the DistributeFees tests.
Not worth it to look into why the testApp integration runs into this problem